build.gradle

```diff
@@ -68,22 +71,24 @@ subprojects {
     showViolations = true
   }
 
-  apply plugin: 'pmd'
-  pmd {
-    ruleSetFiles = files("$rootDir/config/pmd/ruleset.xml")
-    toolVersion = libs.versions.pmd.get()
-    consoleOutput = true
-    ignoreFailures = false
-  }
+  if (JavaVersion.current() < JavaVersion.VERSION_24) {
```

Comment:
We cannot have these plugins for Java 24?

Reply:
Yes, these two plugins doesn't support Java 24 yet.

The development speed is not the same~

Reply:
BTW, these tools (PMD and spotbugs) are static code analysis tools. So, Java 17/21 test coverage are enough already. We don't need them in Java 24.